test/strategies/douglas_peucker.cpp

```diff
@@ -221,12 +225,15 @@ inline void test_with_strategy()
 
     typedef bg::model::point<CoordinateType, 2, bg::cs::cartesian> point_type;
     typedef bg::model::linestring<point_type> linestring_type;
+    typedef bg::model::segment<point_type> segment_type;
     typedef test_one_case<linestring_type> tester;
 
     linestring_type ls1, ls2;
     linestring_type res1, res2;
+    point_type const p1(-6,-13), p2(0,-15);
+    segment_type const s(point_type(12,-3), point_type(-12,5));
 
-    if (boost::is_same<CoordinateType, long double>::value)
+    if (bg::comparable_distance(p1, s) >= bg::comparable_distance(p2, s))
     {
         tester::apply("LINESTRING(12 -3, 4 8,-6 -13,-9 4,0 -15,-12 5)",
                       10,
```

Comment:
I don't understand this case. In SQL Server this linestring is not simplified until the distance is about 13.6
If so, the output is LINESTRING (12 -3, -6 -13, -9 4, 0 -15, -12 5) so it is 4 8 which is removed here. So how can we have for our case both points removed, in one case, and not in another case?
Don't know the exact algorithm, but it is at least suspicious.